### PR TITLE
Refactoring Error Handling

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?
@QuarpT noted in https://github.com/guardian/gateway/pull/638#discussion_r668628677 that it would be useful to log the stack trace of a specific error.

However I noticed that we were using the `logger.error` method wrong the entire time. In https://github.com/guardian/gateway/blob/496851c636f1d1b17f9ea341fcc78b9dfce973fb/src/server/lib/logger.ts#L14 the first parameter is a message, and the second the error. However we were passing the error as the first argument, which was just sent as a useless string.

This PR refactors and adds errors throughout the project with a useful message, and the error in the correct parameter.

It also removes unnecessary try catch blocks too.

### CODE REVIEW

File: src/server/lib/idapi/changePassword.ts [hunk 1]
```diff
@@ -67,6 +68,10 @@ export async function change(password: string, token: string, ip: string) {
     );
     return result.cookies;
   } catch (error) {
+    logger.error(
+      `IDAPI Error changePassword change ${ApiRoutes.CHANGE_PASSWORD}`,
```
Comment (oliverlloyd) [inline src/server/lib/idapi/changePassword.ts:72]: Are we not using `url` here because the `qs` string causes issues?

Comment (oliverlloyd) [reply src/server/lib/idapi/changePassword.ts:72]: If so, maybe we can encode it?

**Review by oliverlloyd — APPROVED** [review]
This looks great!

One question: What will happen if someone does `logger.error(e)`? I can see this being something people might expect the api to support

**Comment by coldlink** [comment]
> This looks great!
> 
> One question: What will happen if someone does `logger.error(e)`? I can see this being something people might expect the api to support

this is what's happening at the moment, whats logged out is something like `{ "message": "Error" }` which is super unhelpful

**Pull request merged**